### PR TITLE
[DEV-11491] Feature - Implement "Placeholders" and dedicated "Menu Item" for Dropbox

### PR DESCRIPTION
**Pull request by MohammadxAli**
## Menu Item

image

## Placeholders

image

image

### CODE REVIEW

File: packages/slate-editor/src/modules/editor/menuOptions.tsx [hunk 1]
```diff
@@ -274,6 +275,18 @@ function* generateOptions(
         };
     }
 
+    if (withEmbeds && withSpecificProviderOptions) {
+        yield {
+            action: MenuAction.ADD_DROPBOX,
+            icon: Icons.ComponentDropbox,
+            group: Group.SPECIFIC_EMBEDS,
```
Comment (MohammadxAli) [inline packages/slate-editor/src/modules/editor/menuOptions.tsx:282]: I will address the new group hierarchy at the end (once all new menu options are added)

There is [DEV-11512](https://linear.app/prezly/issue/DEV-11512/rearrange-the-groups-to-match-the-new-categories) to address that.

File: packages/slate-editor/src/extensions/placeholders/elements/EmbedPlaceholderElement.tsx [hunk 2]
```diff
@@ -101,21 +108,32 @@ export function EmbedPlaceholderElement({
...
+    if (element.provider === PlaceholderNode.Provider.DROPBOX) {
+        return render({
+            inputTitle: 'Dropbox embed',
+            inputDescription: 'Paste a link and hit Enter',
+            inputPlaceholder: 'https://www.dropbox.com/s/file/',
+            inputAction: 'Add embed',
+            title: (props) => <Title {...props} text="Click to insert a Dropbox embed" />,
+            description: (props) => <Description {...props} text="Add using a social media link" />,
+        });
+    }
```
[the remark this answers is not in the document]
Comment (e1himself) [reply packages/slate-editor/src/extensions/placeholders/elements/EmbedPlaceholderElement.tsx:135]: @MohammadxAli After thinking more about it, I think this still should be using the Embed Placeholder. We'll deal with re-routing separately within the scope of DEV-7592.

**Review by e1himself — APPROVED** [review]
Good to merge!

File: packages/slate-editor/src/extensions/placeholders/elements/EmbedPlaceholderElement.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (MohammadxAli) [reply packages/slate-editor/src/extensions/placeholders/elements/EmbedPlaceholderElement.tsx:135]: Alright 👍 Let's keep it this way for now.

**Pull request merged**